striker-ui/Makefile.am

```diff
@@ -1,136 +1,66 @@
+outimagesdir		= pngs
 
-htmldir			= $(localstatedir)/www/html
-outdir			= out
+# The build output directory listed under EXTRA_DIST contains the files to be
+# installed; there is no build process required.
+#
+EXTRA_DIST		= $(nextoutdir)
 
-# Trigger build target on make call without parameters.
-all: out
+# Blank target; the build output directory should be usable as-is.
+$(nextoutdir): ;
 
-# Note: the input file to the generate endpoint prefix step must exist in
-#       EXTRA_DIST.
+# This target is for maintainers only; do not execute during CI/CD.
 #
-out:
-	-@echo "Copying required build files to build (current) directory."
-	rsync -av --exclude "[Mm]ake*" $(srcdir)/ ./
-	-@echo "Allow tools to write to files in the build directory."
-	chmod -R +w .
+$(nodemodulesdir):
 	-@echo "Install node modules (dependencies) prior to building."
 	npm install --no-package-lock --ignore-scripts
-	-@echo "Generate endpoint prefix."
-	sed 's@=.*@=/cgi-bin@' <env.development >.env.local
-	-@echo "Build front-end project."
+
+# This target is for maintainers only; do not execute during CI/CD.
+#
+# Note: this target must **not** be renamed to the same name as the build
+# output directory because it will trigger a build during the `make` implicit
+# call.
+#
+.PHONY: build
+build: clean-build $(nodemodulesdir)
+	-@echo "Build web UI."
 	npm run build
+	-@echo "Stage only the build output directory."
+	git reset && git add $(nextoutdir)
+	-@echo "Commit the build output."
+	git commit -m "chore: rebuild web UI"
+
+# This target is for maintainers only; do not execute during CI/CD.
+#
+# Note: this target is strictly used in case of rebuilding; use
+# maintainer-clean instead to restore the repo to a clean state.
+.PHONY: clean-build
+clean-build:
+	-@echo "Clean up build output files to prepare for rebuild."
+	rm -rf $(nextoutdir) $(nextbuilddir)
 
 install-data-hook:
 	-@echo "Place build output files."
-	cp -r $(outdir)/index.html $(outdir)/_next $(DESTDIR)/$(htmldir)/
+	cp -r --no-preserve=mode $(srcdir)/$(nextoutdir)/$(outindexfile) $(srcdir)/$(nextoutdir)/$(outjsmodulesdir) $(DESTDIR)/$(htmldir)/
 	-@echo "Create symlink to images to enable borrowing icon etc. without duplicating."
-	(cd $(DESTDIR)/$(htmldir); $(LN_S) skins/alteeve/images pngs)
+	(cd $(DESTDIR)/$(htmldir); $(LN_S) skins/alteeve/images $(outimagesdir))
 
 uninstall-hook:
 	-@echo "Remove all installed files of the current module."
-	(cd $(DESTDIR)/$(htmldir); rm -rf index.html _next pngs)
-
-clean-local:
-	-@echo "Clean up build output files."
-	test $(srcdir) == . && rm -rf $(outdir) .next || find . -mindepth 1 -maxdepth 1 -exec rm -rf {} +
+	(cd $(DESTDIR)/$(htmldir); rm -rf $(outindexfile) $(outjsmodulesdir) $(outimagesdir))
 
-distclean-local: clean-local
+maintainer-clean-local: clean-build
+	-@echo "Clean up node modules."
+	rm -rf $(nodemodulesdir)
+	-@echo "Since the build output directory is removed, restore it to keep local repo clean."
+	git restore $(nextoutdir)
```

Comment:
Those 2 won´t work as-is. Why is the build output directory removed in the first place? It shouldn´t on clean or maintainer clean.

The call to git restore cannot be used here or in clean. Only on isolated target as build.

You might be running those commands on a release tarball that have no git information or even git installed and those commands would fail.

Reply:
`git` calls are removed from all clean local hooks.

Removing unused files in `out/` after build is no longer necessary because they be set in .gitignore; this works since the build target is only used in an environment where `git` is available.